services/src/handlers/projects.rs

```diff
@@ -35,6 +38,44 @@ where
         );
 }
 
+mod error {
+    use actix_web::{HttpResponse, ResponseError};
+    use snafu::prelude::*;
+    use strum::IntoStaticStr;
+
+    use crate::{handlers::ErrorResponse, projects::error::ProjectDbError};
+
+    #[derive(Debug, Snafu, IntoStaticStr)]
+    #[snafu(visibility(pub(crate)))]
+    #[snafu(context(suffix(false)))]
+    pub enum ProjectHandlerError {
+        #[snafu(display("Could not create project: {source}"))]
+        CreateProject { source: ProjectDbError },
+        #[snafu(display("Could not list projects: {source}"))]
+        ListProjects { source: ProjectDbError },
+        #[snafu(display("Could not update project: {source}"))]
+        UpdateProject { source: ProjectDbError },
+        #[snafu(display("Could not delete project: {source}"))]
+        DeleteProject { source: ProjectDbError },
+        #[snafu(display("Could not load project version: {source}"))]
+        LoadProjectVersion { source: ProjectDbError },
+        #[snafu(display("Could not load latest project version: {source}"))]
+        LoadLatestProjectVersion { source: ProjectDbError },
+        #[snafu(display("Could not list project versions: {source}"))]
+        ListProjectVersions { source: ProjectDbError },
+    }
+
+    impl ResponseError for ProjectHandlerError {
+        fn status_code(&self) -> actix_http::StatusCode {
+            actix_http::StatusCode::INTERNAL_SERVER_ERROR
```

Comment:
If the project error is  "Could not delete project since the ID does not exist", it is more of a 4xx error, right?

Reply:
yes, but we can not know what the reason for not being able to delete the project is without inspecting the source. In the end I think there is not much benefit in using different error codes. I was once a fan of bein as RESTful as possible, but in the end our REST API is only Json via Http...

Reply:
isn't it a layer breach if we encode the http code of errors for the Db-Errors which do not really need to know anything about web handlers (other than in error case)?

My suggestions would be to just use error code 500 for everything from now on. We could also get rid of all the tests that check for super specific http codes like invalid media type etvc. But we can also discuss this in the team.

Reply:
If this is a breach, it is a breach now, since you've implemented `ResponseError` for an error enum.

You could match against the errors inside the `status_code` fn. It is not super critical, but a pet peeve in our overhauled error handling :shrug: .

Reply:
I don't really think it is a "breach" because it doesn't actually break anything, because AFAIK we do not use the error code anywhere. I still think we should simplify the error handling and disregard the http codes, but here you go:

6642f579498b322ff5dfd7c5a5bcb6f45a645fa4